### ISSUE TITLE
[Feature request] Pass body and query as kwargs

### ISSUE THREAD

**Issue by adriencaccia** [bauerji/flask_pydantic#12]
Hello, I would like to discuss the following feature:

Having the ability to have the potential `query` and `body` objects directly passed as `kwargs` to the method.

The two reasons I like this feature:
1. The code is easier to understand, you do not have to go through the `request` object.
2. If you add the type hints to the arguments (as shown in the example below), `query` and `body` will be correctly typed in your IDE. This brings a much better developer experience.

In the example below, the `validate_and_parse` is doing the same thing as `validate` would except it populates the decorated function `kwargs`.

```python
...
from flask_pydantic import validate_and_parse

...

class QueryModel(BaseModel):
    age: int


class BodyModel(BaseModel):
    name: str
    nickname: Optional[str]


class ResponseModel(BaseModel):
    id: int
    age: int
    name: str
    nickname: Optional[str]


@app.route("/", methods=["POST"])
@validate_and_parse(body=BodyModel, query=QueryModel)
def post(query: QueryModel, body: BodyModel):
    """
    Basic example with both query and body parameters, response object serialization.
    """
    # save model to DB
    id_ = 2

    return ResponseModel(
        id=id_,
        age=query.age,
        name=body.name,
        nickname=body.nickname,
    )
```

If you are interested in this feature, I would really love to have a crack at it.
I already have a quick implementation ready.

**Comment by bauerji** [comment]
Hi,

To be honest I was thinking about this possibility while building this library but at the end I decided to use `request` object since it felt more "flasky" way to me.

However your suggestion makes sense and I would love to have this feature built in. Especially the hints in IDEs would make such a really difference!

Thank You! You are very welcome to implement these changes.

Jirka

**Comment by adriencaccia** [comment]
I am glad that you like this feature :smile: !

I have several ideas regarding the implementation of the feature.

## Solution 1

```python
@app.route("/", methods=["POST"])
@validate_and_parse(body=BodyModel, query=QueryModel)
def post(body: BodyModel, query: QueryModel):
	...
```

The one I presented in my original message. It would imply to:
- Have a new decorator `validate_and_parse`, which would make the API more complex
- Duplicate the `body` and `query` type definition (both in the decorator kwargs and the decorated function args)

## Solution 2

```python
@app.route("/", methods=["POST"])
@validate()
def post(body: BodyModel, query: QueryModel):
	...
```

This solution implies:
- Use the same decorator `validate`, making the API simpler
- The typing of `body` and `query` will be retrieved in `validate` using `func.__annotations__`
- No duplication of parameters definition

We would raise an error if both methods of defining the parameters are used, eg:

```python
@app.route("/", methods=["POST"])
@validate(body=BodyModel, query=QueryModel)
def post(body: BodyModel, query: QueryModel):
	...

# Will throw
```

## Solution 3

```python
@app.route("/", methods=["POST"])
@validate_and_parse()
def post(body: BodyModel, query: QueryModel):
	...
```

Same as last one, except we use a new `validate_and_parse` instead of adding the feature to `validate`.


I would personnaly be more inclined to implement the solution 2.

What do you think ?

**Comment by bauerji** [comment]
This feature is included in package version `0.3.0`. Thank You for contribution!
